client/rpc-api/src/dev/mod.rs

```diff
@@ -0,0 +1,62 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Substrate dev API.
+
+pub mod error;
+
+use self::error::Result;
+use codec::{Decode, Encode};
+use jsonrpc_derive::rpc;
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+
+/// Statistics of a block returned by the `dev_getBlockStats` RPC.
+#[derive(Eq, PartialEq, Clone, Copy, Encode, Decode, Debug, TypeInfo, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BlockStats {
+	/// The length in bytes of the storage proof produced by executing the block.
+	pub witness_len: u64,
+	/// The length in bytes of the storage proof after it was compacted.
+	pub witness_compact_len: u64,
+	/// Length of the block in bytes.
+	///
+	/// This information could also be acquired by downloading the whole block. This is merely
+	/// a convenience to save some complexity on the client side.
+	pub block_len: u64,
+	/// Number of extrinsics in the block.
+	///
+	/// This information could also be acquired by downloading the whole block. This is merely
+	/// a convenience to save some complexity on the client side.
+	pub num_extrinsics: u64,
+}
+
+/// Substrate dev API.
+///
+/// This API contains unstable and unsafe methods only meant for development nodes. They
+/// are all flagged as unsafe for this reason.
+#[rpc]
+pub trait DevApi<Hash> {
+	/// Reexecute the specified `block_hash` and gather statistics while doing so.
+	///
+	/// This function will require the specified block and its parent to be available
+	/// at the queried node. If either the specified block or the parent are not available,
+	/// this function will return `None`.
```

Comment:
```suggestion
	/// This function requires the specified block and its parent to be available
	/// at the queried node. If either the specified block or the parent is pruned,
	/// this function will return `None`.
```

Is there any other reason than pruning that a block is unavailable?